siddhi-test-framework/src/main/java/io/siddhi/distribution/test/framework/SiddhiRunnerContainer.java

```diff
@@ -209,18 +214,43 @@ private int getStartupTimeoutSeconds() {
     @Override
     protected void waitUntilContainerStarted() {
         logger().info("Waiting for Siddhi Runner Container to start...");
-        retryUntilSuccess(getStartupTimeoutSeconds(), TimeUnit.SECONDS, () -> {
-            if (!isRunning()) {
-                throw new ContainerLaunchException("Siddhi Runner Container failed to start");
-            }
-            HTTPClient.HTTPResponseMessage httpResponseMessage = callHealthAPI();
-            if (httpResponseMessage.getResponseCode() == 200) {
-                logger().info("Siddhi Runner Health API reached successfully.");
-                return null;
-            } else {
-                throw new ConnectException("Failed to connect with the Siddhi Runner health API");
+        if (deploymentDirectoryFile != null) {
+            String[] siddhiAppsArray = deploymentDirectoryFile.list();
+            assert siddhiAppsArray != null;
+            for (String siddhiApp : siddhiAppsArray) {
+                String fileName = siddhiApp.substring(0, siddhiApp.length() - 7);
```

Comment:
Instead of having hardcoded 7; use ".siddhi".length(). You can have  ".siddhi" as a constant as well for ease

Reply:
Fixed in https://github.com/siddhi-io/distribution/pull/816/commits/684ff2789b8193d4b3820859614a6de441dbe914